src/software/jetson_nano/thunderloop.cpp

```diff
@@ -416,6 +417,34 @@ double Thunderloop::getCpuTemperature()
     }
 }
 
+bool isPowerStable(std::ifstream& log_file)
+{
+    // if the log file cannot be open, we would return false. Chances are, the battery
+    // power supply is indeed stable
+    if (!log_file.is_open())
+    {
+        LOG(WARNING) << "Cannot dmesg log file. Do you have permission?";
+        return true;
+    }
+
+    std::string line;
+    while (std::getline(log_file, line))
+    {
+        // if this lines exist, we know for sure that the battery is not stable!
+        if (line.find("soctherm: OC ALARM 0x00000001") != std::string::npos)
+        {
+            return false;
+        }
+    }
+
+    // https://pastebin.com/ebsATa2h we need to clear the error state bit because we have
+    // reached the end of the file. After being clearing the bit, std::getline would
+    // return the changes in the file. See the pastebin for example.
+    log_file.clear();
```

Comment:
I'm not sure if I understand what's happening here. What is the error state bit?

Also
```suggestion
    // https://pastebin.com/ebsATa2h we need to clear the error state bit because we have
    // reached the end of the file. After clearing the bit, std::getline would
    // return the changes in the file. See the pastebin for example.
    log_file.clear();
```

Reply:
I think it's because we read the log file to completion and `std::getline` will set an error flag once it encounters EOF

if we want to reuse the file descriptor, we need to clear this flag before trying to read more

Reply:
something like that. I've updated the comments. I hope this makes more sense.